presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java

```diff
@@ -67,7 +68,6 @@
 import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
 import static io.airlift.concurrent.MoreFutures.addTimeout;
 import static io.airlift.concurrent.Threads.threadsNamed;
-import static io.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
```

Comment:
Should we blacklist this method (eg using modernizer)?

Reply:
Note that apparently there are more places like this. 

Also, we could have a test for this -- you can ask docker for a specific hostname

Reply:
I think we should update it to fail if passed a URL without a host.

Is there an existing test we could update to use a different host name? We could have one with underscore and dashes.

Reply:
You can take `TestServer` and override `Host` header.

if you want the _real_ hostname to have an underscrore, you need docker.
In docker (compose) (or PT Launcher), hostname = container name, so existing envs don't have an underscore and cannot be changed to have one. You'd need a new one.